sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -1245,3 +1245,27 @@ case class BRound(child: Expression, scale: Expression)
     with Serializable with ImplicitCastInputTypes {
   def this(child: Expression) = this(child, Literal(0))
 }
+
+/**
+ * The number truncated to scale decimal places.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(number, scale) - Returns number truncated to scale decimal places. " +
+    "If scale is omitted, then number is truncated to 0 places. " +
+    "scale can be negative to truncate (make zero) scale digits left of the decimal point.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(1234567891.1234567891, 4);
+       1234567891.1234
+      > SELECT _FUNC_(1234567891.1234567891, -4);
+       1234560000
+      > SELECT _FUNC_(1234567891.1234567891);
+       1234567891
+  """)
+// scalastyle:on line.size.limit
+case class Truncate(child: Expression, scale: Expression)
```

Comment:
I am still preferring to extend `trunc`.  Not straightforward to know the difference between `truncate` and `trunc`

Reply:
Hi @wangyum, can you still extend `trunc`?
If not, what was the major reason you decided to separate these? Thanks!

Reply:
For example: `trunc(StringType)`, We don't know we should trunc to number or trunc to date.